MS-125 Added ay-accordion and ay-accordion-header

tkalra opened the pull request:
- Updated target to es6
- Removed angular ay-accordion
- added rollup configurations
- added ay-accordion web component
- added ay-accordion-header web component

Refs MS-125

Commit:
MS-125 Added ay-accordion and ay-accordion-header

- Updated target to es6
- Updated angular ay-accordion to es6
- added rollup configurations
- added ay-accordion web component
- added ay-accordion-header web component

Refs MS-125

dpogue commented on src/web-component/ay-accordion-header/index.ts:
```diff
@@ -0,0 +1,68 @@
+
+    const toggleOnClick = () => {
+      let toggleEvent = new Event('toggle');
+      console.log('dispatching click event from header');
```
Stray console.log

dpogue commented on src/web-component/ay-accordion-header/index.ts:
```diff
@@ -0,0 +1,68 @@
+      let toggleEvent = new Event('toggle');
+      if (event.keyCode === 32 || event.keyCode === 13) {
+        ayAccordionElem.dispatchEvent(toggleEvent);
+      } else {
```
I don't think we need this `else` block

dpogue commented on src/web-component/ay-accordion-header/index.ts:
```diff
@@ -0,0 +1,68 @@
+    const toggleOnClick = () => {
+      let toggleEvent = new Event('toggle');
+      console.log('dispatching click event from header');
+      ayAccordionElem.dispatchEvent(toggleEvent);
```
1. It would be better to look up the closest ayAccordion element every time we dispatch an event, in case stuff has moved around in the DOM.

2. It's valid to just use `el.dispatchEvent(new Event('toggle'))`, although you might need to handle older browsers where you'd need to use `document.createEvent('toggle')` instead

dpogue commented on src/web-component/ay-accordion-header/index.ts:
```diff
@@ -0,0 +1,68 @@
+
+    if (accordionHeaderClickMap.has(this)){
+      this.removeEventListener('click', accordionHeaderClickMap.get(this));
+    }
```
We should remove `this` from the maps when we remove the event listeners too

Commit:
MS-125 Removed console.log

Refs MS-125

dpogue commented on src/web-component/ay-accordion/index.ts:
```diff
@@ -0,0 +1,104 @@
+    }
+  }
+
+  //Do we need to pass 'this' to it ?
```
> Do we need to pass 'this' to it ?

No, this is a getter for the `open` property

We *should* have a setter as well, that takes a boolean and calls either `toggleOpen` or `toggleClose`

dpogue commented on src/web-component/ay-accordion/index.ts:
```diff
@@ -0,0 +1,104 @@
+
+  disconnectedCallback() {
+    if (accordionEventMap.has(this)){
+      this.removeEventListener('toggle', accordionEventMap.get(this));
```
Remove `this` from the map after unregistering

dpogue commented on src/web-component/ay-accordion/index.ts:
```diff
@@ -0,0 +1,104 @@
+    });
+
+    if(this.hasAttribute('disabled')) {
+      this.setAttribute('aria-disabled', 'true');
```
`aria-disabled` *probably* belongs on the accordion-header, since that's the bit exposed as a button

Commit:
MS-125 Added setter for open property
 - Removed webComponent from the weakMaps on disconnection
 - Refactored toggleOnClick and OnPress methods

Refs MS-125

Commit:
MS-125 Added the aria-disabled attr to header comp

Refs MS-125

dpogue commented on src/web-component/ay-accordion-header/index.ts:
```diff
@@ -0,0 +1,69 @@
+    this.setAttribute('role', 'button');
+    this.setAttribute('tabIndex', '0');
+
+    if(this.closest('ay-accordion').hasAttribute('disabled')) {
```
It might be easier to have the accordion find its child header and set the aria-disabled attribute. The way you're doing it right now won't dynamically add/remove the attribute if the accordion is enabled or disabled dynamically.

Commit:
MS-125 Added attribute change callbacks
- Added observed variables to ay-accordion
- Setting aria-disabled on ay-accordion-header

Refs MS-125

Commit:
MS-125 Removed toggle functions

Refs MS-125

Commit:
MS-125 Refactored handle toggle

Refs MS-125

dpogue committed:
Merge pull request #20 from AyogoHealth/attribute-change

MS-125 Added attribute change callbacks

Commit:
MS-125 Added eslint

Refs MS-125

dpogue committed:
Merge pull request #21 from AyogoHealth/add-eslint

MS-125 Added eslint

Commit:
MS-125 Added ay-accordion-root
- Added animations for toggle
Refs XX-000

Commit:
MS-125 Added comments for ay-accordion-root
- Changed time of animation to 1s

Refs MS-125

dpogue committed:
Merge pull request #22 from AyogoHealth/ay-accordion-root

MS-125 Added ay-accordion-root

Commit:
MS-125 Added Documentation page for ay-accordion
- Refactored mulitple attr handling

Refs MS-125

Commit:
MS-125 Update demo styles
- edited the link to dist file
Refs MS-125

Commit:
Merge pull request #23 from AyogoHealth/update-demo

MS-125 Added documentation page for web component based ay-accordion

Commit:
MS-125 Remove Angular dependency

Refs MS-125

Commit:
MS-125 Removing angualr peer dependency

Refs MS-125

Commit:
MS-125 Renamed web-component html to index.html

Refs MS-125

Commit:
MS-125 Updated README.md
- Updated package.json
   - Added author
   - Combined dist and build scripts
- Added accordion-root component to index file
- Updated demo page

Refs MS-125

dpogue commented:
Is there a reason we're putting the output in `dist/web-component/index.js` instead of `dist/index.js`?

Commit:
MS-125 Changed output of rollup to dist/index

Refs MS-125

dpogue reviewed (approved):
:+1:
